### PR TITLE
kiraarghy:rotation-fix-iphone

### PR DESCRIPTION
**Pull request by kiraarghy**
In response to issue #9639.

Bug:

1. I go to my map on iPhone safari/ chrome.
2. I move my iPhone to landscape.
3. I move my iPhone to portrait.
4. The map does not resize to fit the portrait layout.

Added to map.js a screen orientation change event listener, this will trigger the `_windowResize method` so that when used on devices, orientation changes will trigger a resize. 

Unfortunately, the `orientationchange` event listener seems to be [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Window/orientationchange_event#Browser_compatibility) so to ensure future compatibility I have carried out this fix with the [Screen.Orientation API](https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/onchange) with a fallback to the deprecated `orientationchange` event for browsers which don't support the Screen.Orientation API.

I've user tested this locally on a simulated and using a friend's iPhone and this seems to work but additional user testing would be very welcome.

I'm having some trouble getting my head around how testing for this project works, could someone assist me in that?

Thanks!

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fix for iPhone orientation change not triggering map resize by adding event listener for orientation change.</changelog>`

### CODE REVIEW

File: src/ui/map.js [hunk 1]
```diff
@@ -436,6 +436,13 @@ class Map extends Camera {
         if (typeof window !== 'undefined') {
             window.addEventListener('online', this._onWindowOnline, false);
             window.addEventListener('resize', this._onWindowResize, false);
+            if (typeof window.screen.orientation === 'object') {
+                // ensures browser supports latest ScreenOrientation API
+                window.screen.orientation.addEventListener('change', this._onWindowResize, false);
```
Comment (andrewharvey) [inline src/ui/map.js:441]: wherever an event listener is added on map create, it needs to also be removed on map remove, eg https://github.com/mapbox/mapbox-gl-js/blob/42ccea03b45d147628f261bca62c27fdace5eaa1/src/ui/map.js#L2566-L2569

**Comment by kiraarghy** [comment]
I was looking into the issue @Equanimaster pointed out about the possibility of more resizing events being dispatched but in my profiling, this seems to result in the same number of calls to `_onWindowResize` as occurs in master currently.

Master:
Screenshot 2020-05-06 at 16 00 22

My branch: 
Screenshot 2020-05-06 at 16 07 26

I think left to do is the benchmark scores and tests, @andrewharvey can you help at all with the testing?

**Comment by kiraarghy** [comment]
Hey is there any chance someone could have a look at this PR and help out?

File: src/ui/map.js [hunk 2]
```diff
@@ -436,6 +436,13 @@ class Map extends Camera {
...
+            } else {
+                // If not then use deprecated 'orientationchange' event
+                window.addEventListener('orientationchange', this._onWindowResize, false);
```
Comment (mourner) [inline src/ui/map.js:444]: As far as I can see, `window` `orientationchange` event [is a part of a living standard](https://developer.mozilla.org/en-US/docs/Web/API/Window/orientationchange_event), it's not deprecated. So you can get rid of the branching and only use this one. Perhaps it was mistaken for [screen orientationchange](https://developer.mozilla.org/en-US/docs/Web/API/Screen/onorientationchange), which is indeed deprecated?

Comment (kiraarghy) [reply src/ui/map.js]: Yep mistake on my part, have included this change in my last commit.

**Pull request merged**